### PR TITLE
XWIKI-22211: Deprecate NotificationFilterPreferenceProvider

### PR DESCRIPTION
**Pull request by surli**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-22211

# Changes

## Description

  * Deprecate and legacify NotificationFilterPreferenceProvider and
    NotificationFilterPreference#getProviderHint
  * Clean up all code relying on it simplifying a lot the implem of
    NotificationFilterPreference
  * Only valid usage of the providerHint was for checking if a filter
    preference was stored or not: replaced it with checking the ID of
the filter

## Clarifications

N/A

# Executed Tests

`mvn clean install -Pquality,integration-tests,docker` on both `xwiki-platform-notifications` and `xwiki-platform-legacy-notifications`.

# Expected merging strategy

* Prefers squash: Yes
* Backport on branches:
  * None

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterExpressionGenerator.java [hunk 1]
```diff
@@ -297,8 +295,7 @@ private boolean isEnabledScopeNotificationFilterPreference(NotificationFilterPre
     {
         // This optimization can only works on preferences stored by the user, that's why we add a condition
         // on the provider hint.
-        return nfp.isEnabled() && ScopeNotificationFilter.FILTER_NAME.equals(nfp.getFilterName())
-                && UserProfileNotificationFilterPreferenceProvider.HINT.equals(nfp.getProviderHint());
+        return nfp.isEnabled() && ScopeNotificationFilter.FILTER_NAME.equals(nfp.getFilterName());
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterExpressionGenerator.java:298]: Hmmm I just realized that this method was having the following documentation:
> NOTE: this work only with user's preferences stocked in the database. It does not support the use-case of
     * the notifications macro, when the preferences are created on the fly based on the macro parameters!

I probably need to check it a bit more.

Comment (surli) [reply xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterExpressionGenerator.java]: I double checked it and the issue was about being able to check if the filter was stored or created on the fly: I decided to go for a check on the filter ID for this, since we use some internal prefixes for storing preferences.

**Comment by surli** [comment]
Switching to draft as there's apparently a regression when executing integration tests.

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterPreferenceManager.java [hunk 2]
```diff
@@ -19,266 +19,123 @@
...
-import org.xwiki.notifications.filters.NotificationFilterPreferenceProvider;
 import org.xwiki.notifications.filters.NotificationFilterType;
 
 /**
- * Default implementation of {@link NotificationFilterPreferenceManager}.
+ * Default implementation of the role {@link NotificationFilterPreferenceManager}.
  *
  * @version $Id$
- * @since 10.9
+ * @since 9.8RC1
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterPreferenceManager.java:45]: Is this change on purpose ?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-notifications/xwiki-platform-legacy-notifications-filters/xwiki-platform-legacy-notifications-filters-api/pom.xml [hunk 3]
```diff
@@ -37,14 +37,18 @@
          By setting this property, we make sure this application would be automatically uninstalled if the administrator
          installs the old Activity Stream Application. -->
     <xwiki.extension.features>org.xwiki.platform:xwiki-platform-notifications-filters-api</xwiki.extension.features>
-    <xwiki.jacoco.instructionRatio>0.63</xwiki.jacoco.instructionRatio>
+    <xwiki.jacoco.instructionRatio>0.08</xwiki.jacoco.instructionRatio>
   </properties>
   <dependencies>
+    <!-- Trigger xwiki-platform-notifications-fomters-api dependencies (but without
```
[the remark this answers is not in the document]
Comment (surli) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-notifications/xwiki-platform-legacy-notifications-filters/xwiki-platform-legacy-notifications-filters-api/pom.xml:43]: Yep same diagonal on the keyboard for om and il it seems

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/test/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterTest.java [hunk 4]
```diff
@@ -268,7 +267,6 @@ void withTopLevelInclusiveFilters() throws Exception
         DocumentReference documentReference = new DocumentReference("wikiA", "SpaceM", "DocumentN");
         NotificationFilterPreference prefζ = mockNotificationFilterPreference("wikiA:SpaceM.DocumentN",
                 documentReference, NotificationFilterType.INCLUSIVE, null);
-        when(prefζ.getProviderHint()).thenReturn("userProfile");
         when(prefζ.getStartingDate()).thenReturn(new Date(99000));
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/test/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterTest.java:270]: Not something introduced by this pull request by that's a pretty surprising character to use in a variable name.

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterPreferenceManager.java (hunk 2 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterPreferenceManager.java:45]: Probably not, I'm not sure how I ended up changing this...
